pinot-core/src/main/java/org/apache/pinot/core/segment/creator/impl/SegmentColumnarIndexCreator.java

```diff
@@ -216,6 +225,17 @@ public void init(SegmentGeneratorConfig segmentCreationSpec, SegmentIndexCreatio
             .put(columnName, new LuceneTextIndexCreator(columnName, _indexDir, true /* commitOnClose */));
       }
 
+      if (_fstIndexColumns.contains(columnName)) {
+        Preconditions.checkState(fieldSpec.isSingleValueField(),
+            "FST index is currently only supported on single-value columns");
+        Preconditions.checkState(fieldSpec.getDataType() == DataType.STRING,
+            "FST index is only supported on STRING type columns");
+        Preconditions.checkState(indexCreationInfo.isCreateDictionary(),
```

Comment:
isCreateDictionaryForColumn() is the correct method to be used for checking if dictionary is to be created or not. indexCreationInfo.isCreateDictionary() will always return true currently

Reply:
Done